src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php

```diff
@@ -244,16 +244,18 @@ public function prepare_item_for_response( $item, $request ) {
 		// Wrap the data in a response object.
 		$response = rest_ensure_response( $data );
 
-		$response->add_links(
-			array(
-				'collection'              => array(
-					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
-				),
-				'https://api.w.org/items' => array(
-					'href' => rest_url( rest_get_route_for_post_type_items( $post_type->name ) ),
-				),
-			)
-		);
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$response->add_links(
+				array(
+					'collection'              => array(
+						'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+					),
+					'https://api.w.org/items' => array(
+						'href' => rest_url( rest_get_route_for_post_type_items( $post_type->name ) ),
+					),
+				)
+			);
```

Comment:
I think we should take the opportunity to break this logic out to `prepare_links` method.